### PR TITLE
Update the NEWS file

### PR DESCRIPTION
**Pull request by jkbonfield**
*Phew*!

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,231 +1,444 @@
...
+   works when bgzipped.)
+
+ * HTSlib now supports the @SQ-AN header field, which lists alternative names
+   for reference sequences.  This means given "@SQ SN:1 AN:chr1", tools like
+   samtools can accept requests for "1" or "chr1" equivalently.  (PR #931)
+
+ * Completely blank files are no longer considered to be a SAM file
+   with no header and no alignments.  Thus pipelines such as `somecmd
+   | samtools ...` with `somecmd` aborting before outputting anything
+   will not propagate the error to the second command. (#261)
```
Comment (jmarshall) [inline NEWS]: s/not/now/!!

But I don't really see why you've rewritten the paragraph entirely. Previous was

```
* Empty (0-length) files are no longer considered to be valid SAM files.
  Input files without any headers or records at all are usually seen in
  pipelines such as `somecmd | samtools ...` with `somecmd` aborting and
  outputting nothing, and it is beneficial for the second command to
  propagate the error.  (#261)
```

This wording was carefully chosen. “Empty (0-length)” is clearer and more precise than “Completely blank”; “blank files … a SAM file” is something of a number mismatch; the user documentation uses both “records” and “alignments” and in this context I think the former is clearer; the line break around the pipe was avoided; the thing that is doing the propagating is the second command itself not some other magic. OTOH _aborting before outputting anything_ is an improvement and “it is beneficial for” was awkward and would benefit from some rewording.

**Comment by jmarshall** [comment]
Formatting nits:

1. The existing entries in this document use `*` in the first column as the top-level bullet and `-` in the third column as a second-level bullet.

   These additions have a mixture of zero, one, or two spaces indentation before the `*` bullet. And some mixture of indentation styles for the second-level `-` too. It would be good to change this to be consistent with the existing style.

2. The new entries are inconsistent as to whether they end in a fullstop or not. Previous convention has been that they do (but the final `(#816)` references don't).

File: NEWS [hunk 2]
```diff
@@ -1,231 +1,444 @@
...
+ * Fixed bug where some 8 or 16-bit negative integers were stored using values
+   reserved by the BCF specification.  These numbers are now promoted to the
+   next size up, so -121 to -128 are stored using at least 16 bits, and -32761
+   to -32768 are stored using 32 bits.
+
+   Note that while BCF files affected by this bug are technically incorrect,
+   it is still possible to read them.  When converting to VCF format,
+   HTSlib (and therefore bcftools) will interpret the values as intended
+   and write out the correct negative numbers. (#766, thanks to John
+   Marshall, samtools/bcftools#874)
```
Comment (jmarshall) [inline NEWS]: Please don't break people's names across lines (cf samtools/samtools#1020).

File: NEWS (hunk 1 above)
Comment (jkbonfield) [reply NEWS]: The reason I rewrote it is mainly because trying to merge changes listed in around 5000 lines of git logs with a couple thousand lines of sporadic NEWs comments, which are all reordered, was just nigh on impossible.

Instead I started again from scratch, and then searched through the old news file to see what was worth lifting over and to check I hadn't missed anything.  I found plenty of cases where I'd rephrased it, but generally prefer brevity over excessive verbiage.  People can look at the git logs if they want the fine detail.  We're not trying to write a specification.  Words like "blank" are fine IMO if it saves extra waffle.

Comment (jmarshall) [reply NEWS]: I appreciate that this was a lot of effort; reviewing it is going to be quite an effort too. Should I assume that the indentation nits will be fixed up before I proofread the whole thing again, or proofread it as is?

Brevity is good of course, as is accuracy and clarity. Where effort has been made over the last 16 months to write clear accurate NEWS entries contemporaneously with the associated changes, it seems unfortunate to discard them.

Comment (daviesrob) [reply NEWS]: Minor formatting problems will be fixed before this is merged.  Unfortunately it might take a bit longer than expected to get on to this as my equivalent updates to samtools just got blown away by a brief power cut. :disappointed:

File: NEWS [hunk 3]
```diff
@@ -1,231 +1,444 @@
...
+
+ * Sped up decoding of CRAM by around 10% when the MD tag is being
+   generated.
+
+ * CRAM REF_PATH now supports %Ns (where N is a single digit)
+   expansion in http URLs, similar to how it already supported this
+   for directories.
+
+ * BGZF now permits indexing and seeking using virtual offsets in
+   completely uncompressed streams.
```
Comment (valeriuo) [inline NEWS]: (#904, thanks to Adam Novak)

**Comment by daviesrob** [comment]
Formatting has been adjusted and links to pull requests added.  Ignore white-space when comparing to make the substantive changes easier to see.

File: NEWS [hunk 4]
```diff
@@ -1,214 +1,392 @@
...
+
+* New API functions bam_set_mempolicy() and bam_get_mempolicy() have
+  been added.  These allow more control over the ownership of bam1_t
+  alignment record data; see documentation in htslib/sam.h for more
+  information. (#922)
+
+* Added more HTS_RESULT_USED checks, this time for VCF I/O. (#805)
+
+* khash can now hash kstrings.  This makes it easier to hash
+  non-nul-terminated strings. (#713)
```
Comment (jmarshall) [inline NEWS]:
```suggestion
  non-NUL-terminated strings. (#713)
```

File: NEWS [hunk 5]
```diff
@@ -1,214 +1,392 @@
...
+* htsFile has a new 'state' member to support SAM multi-threading. (#916)
+
+* A new field has been added to the bam1_t structure, and others
+  have been rearranged to remove structure holes. (#709; #922)
+
+
+Bug fixes
+---------
+
+* Several bgzf format fixes:
```
Comment (jmarshall) [inline NEWS]:
```suggestion
* Several BGZF format fixes:
```

File: NEWS [hunk 6]
```diff
@@ -1,214 +1,392 @@
...
+  supplied fasta file differs to the order of the @SQ headers. (#935)
+
+* Fixed BAM and CRAM multi-threaded decoding when used in conjunction
+  with the multi-region iterator. (#830; #577, #822, #926 all reported by
+  Brent Pedersen)
+
+* Removed some unaligned memory accesses in CRAM encoder and
+  undefined behaviour in BCF reading (#867, thanks to David Seifert)
+
+* Repeated calling of bcf_empty() no longer crash. (#741)
```
Comment (jmarshall) [inline NEWS]:
```suggestion
* Repeated calling of bcf_empty() no longer crashes. (#741)
```

File: NEWS [hunk 7]
```diff
@@ -218,14 +396,86 @@ Noteworthy changes in release a.b
   Note that while BCF files affected by this bug are technically incorrect,
   it is still possible to read them.  When converting to VCF format,
   HTSlib (and therefore bcftools) will interpret the values as intended
-  and write out the correct negative numbers. (#766, thanks to John Marshall)
+  and write out the correct negative numbers. (#766, thanks to John Marshall;
+  samtools/bcftools#874)
+
+* Allow repeated invocations of bcf_update_info and format_*
```
Comment (jmarshall) [inline NEWS]:
```suggestion
* Allow repeated invocations of bcf_update_info() and bcf_update_format_*()
```

**Comment by jmarshall** [comment]
Good work all — that's looking in good shape. I didn't spot any other typos, my only other suggestion would be multiple blank lines to set this epic off from the previous _Noteworthy changes in release 1.9_ heading.

**Comment by daviesrob** [comment]
Thanks, suggestions applied.

File: NEWS [hunk 8]
```diff
@@ -218,14 +396,91 @@ Noteworthy changes in release a.b
   Note that while BCF files affected by this bug are technically incorrect,
   it is still possible to read them.  When converting to VCF format,
   HTSlib (and therefore bcftools) will interpret the values as intended
-  and write out the correct negative numbers. (#766, thanks to John Marshall)
+  and write out the correct negative numbers. (#766, thanks to John Marshall;
+  samtools/bcftools#874)
+
+* Allow repeated invocations of bcf_update_info() and format_*()
```
Comment (jmarshall) [inline NEWS]: Suggestions almost applied :smile: (Original was 8ab9acf01b5fc95c2a6c18ace1e0a489d0089261 “Allow repeated invocations of bcf_update_info/format_*()” but with `/` replaced by `and` it probably needs the full function name prefix for clarity.

```suggestion
* Allow repeated invocations of bcf_update_info() and bcf_update_format_*()
```

Comment (daviesrob) [reply NEWS]: Should be right now.

Comment (jmarshall) [reply NEWS]: :+1: thanks
